src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java

```diff
@@ -155,11 +157,14 @@ public final Checker createChecker(Configuration moduleConfig)
     protected final Checker createChecker(Configuration moduleConfig,
                                     ModuleCreationOption moduleCreationOption)
             throws Exception {
-        final DefaultConfiguration dc;
+        final Configuration dc;
 
         if (moduleCreationOption == ModuleCreationOption.IN_TREEWALKER) {
             dc = createTreeWalkerConfig(moduleConfig);
         }
+        else if (ROOT_MODULE_NAME.equals(moduleConfig.getName())) {
+            dc = moduleConfig;
+        }
```

Comment:
please explain reason of this update , following `else` create RootConfig.

Reply:
Because, otherwise it fails with 'Unable to instantiate "root" class, it is also not possible to instantiate it as "rootCheck" ...'

The problem, when you call `createRootConfig()` method, it creates configuration with name root (`new DefaultConfiguration("root")`).  And then you call `createChecker(Configuration moduleConfig, ModuleCreationOption moduleCreationOption)`, it goes to `else` section and creates `root` configuration with `root` child. So when this configuration is being passed to `Checker` object, it fails, because there is no such module called `root`.

I took this `else if` part from original `AbstractModuleTestSupport`

Reply:
ok, we will think about this in next PR.